src/main/scala/eventstore/EsProjectionsClient.scala

```diff
@@ -0,0 +1,280 @@
+package eventstore
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{ Authorization, BasicHttpCredentials }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.stream.ThrottleMode.Shaping
+import akka.stream.scaladsl.{ Flow, Sink, Source }
+import eventstore.EsProjectionsClient.{ Continuous, OneTime, ProjectionMode, Transient }
+import play.api.libs.json.Json
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Try
+
+/**
+ * The API miss documentation so I used the C# client as a starting point
+ * See : https://github.com/EventStore/EventStore/blob/release-v3.9.0/src/EventStore.ClientAPI/Projections/ProjectionsClient.cs
+ */
+protected[this] trait EventStoreProjectionsUrls {
+  def createProjectionUrl(name: String, mode: ProjectionMode = Continuous, allowEmit: Boolean = true): String = {
+    val emit = if (allowEmit) "emit=1&checkpoints=yes" else "emit=0"
+    val projectionModeStr = mode match {
+      case OneTime    => "onetime"
+      case Transient  => "transient"
+      case Continuous => "continuous"
+    }
+    s"/projections/$projectionModeStr?name=$name&type=JS&$emit"
+  }
+
+  def projectionBaseUrl(name: String): String = s"/projection/$name"
+
+  def fetchProjectionStateUrl(name: String, partition: Option[String]): String =
+    s"${projectionBaseUrl(name)}/state" + partition.fold("")(partition => s"?partition=$partition")
+
+  def fetchProjectionResultUrl(name: String, partition: Option[String]): String =
+    s"${projectionBaseUrl(name)}/result" + partition.fold("")(partition => s"?partition=$partition")
+
+  def projectionCommandUrl(name: String, command: String): String =
+    s"${projectionBaseUrl(name)}/command/$command"
+}
+
+object EsProjectionsClient {
+  sealed trait ProjectionMode
```

Comment:
Makes sense !
Actually, this is kind of a current debate at work on this !
I agree that putting it inside the companion object avoids potential name clash